### ISSUE TITLE
Initial functionality

### ISSUE THREAD

**Issue by davestewart** [davestewart/nuxt-areas#1]
Build

- [ ] convert to TypeScript
- [ ] setup bundling

Configuration

- [x] make `dir` configuration optional (so partial migration doesn't break the site)
- [x] make setting names more explicit

Routes

- [x] build route discovery code
- [x] add route helpers (and move injection of path from service to helper)
- [x] add config files to watch
- [x] add partial route configuration
- [x] add grouping

Additional functionality

- [x] upgrade store discovery to account for grouping
- [x] support additional folders / packages
- [ ] support group-level index page and store

Docs

- [x] split new site setup and migration into separate docs
- [x] configuration options
- [x] routing options
- [x] alias configuration (TS + Webpack)

**Comment by Jamiewarb** [comment]
Hey Dave!
Thought I'd check just before I dive in - does this support Nuxt 3 and Bridge / are you planning to support Nuxt 3 and Bridge, via Nuxt Kit?

**Comment by davestewart** [comment]
Howdy Jamie!

Well, planning to support all of it as soon as I can, and will be open to PRs once the codebase is more stable.

I did install Bridge on a repo the other day but even with Daniel Roe's help couldn't get the whole thing working in the end, but we could try with the [demo](https://github.com/davestewart/nuxt-areas-demo) repo and iron out any bugs.

Is there anything radically different that you worry might not work, or would need extra support?

Obviously to be a success, Areas will need to work everywhere...

**Comment by davestewart** [comment]
Looks like there is a migration guide here:

- https://v3.nuxtjs.org/getting-started/migration/

**Comment by Jamiewarb** [comment]
Nope, nothing that I'd need support on. Just that I'm building a greenfield Nuxt 2 with the aim to add Bridge and eventually Nuxt 3 sometime in December if I can.

Wanted to check what your plans were before I adjust the app for Nuxt Areas, knowing I'd be looking at migration in a months time :)

**Comment by davestewart** [comment]
Would definitely like some more testing done in the wild.

I have always found Nuxt to be frustratingly temperamental.

Areas is very happy on the demo project, then I upgraded a live project from v0.5.0 to  v0.6.0 and it blew up. I don't know if this is Areas code, or Nuxt code right now.

I'm hoping  v0.7.0 will solve things as it brings more structure to the initial pass of determining the areas. I'm trying to add flexible support for nesting, without giving folks enough rope to hang themselves with.

I find when Nuxt blows up it's entirely unforgiving, so I'm hoping to get all these edge cases out the way now so you don't end up staring into the abyss of a Nuxt server side error screen when need it to be working!

**Comment by davestewart** [comment]
Areas 0.7 is out now and it's pretty solid.

**Comment by davestewart** [comment]
@Jamiewarb, see #6 and #7.

It's all good from Nuxt Areas's POV but the Nuxt 2 > 3 migration is no fun. There be dragons!
